compiler/test/fail_compilation/bool_cast.d

```diff
@@ -0,0 +1,14 @@
+/*
+REQUIRED_ARGS: -de
+TEST_OUTPUT:
+---
+fail_compilation/bool_cast.d(12): Deprecation: cast from `ubyte[]` to `bool[]` not allowed in safe code
```

Comment:
(Not that you need to do this right away, but) can we link to/refer users to the spec section/other official documentation, as to _why_ this is an unsafe operation? It was certainly not immediately obvious to me as to why this is the case.

Reply:
I've added a supplemental message for now.